src/main/java/com/devonfw/ide/sonarqube/common/api/config/DevonArchitecturePackage.java

```diff
@@ -1,129 +1,158 @@
-  public String getLayer() {
+  public boolean isLayerClient() {
 
-    return this.devon4jPackage.getLayer();
+    return this.matcher.group(2).matches("client");
   }
 
   public boolean isLayerCommon() {
 
-    return this.devon4jPackage.isLayerCommon();
+    return this.matcher.group(2).matches("common");
   }
 
   public boolean isLayerDataAccess() {
 
-    return this.devon4jPackage.isLayerDataAccess();
+    return this.matcher.group(2).matches("dataaccess");
   }
 
   public boolean isLayerLogic() {
 
-    return this.devon4jPackage.isLayerLogic();
+    return this.matcher.group(2).matches("logic");
   }
 
   public boolean isLayerService() {
 
-    return this.devon4jPackage.isLayerService();
+    return this.matcher.group(2).matches("service");
   }
 
-  public boolean isLayerBatch() {
+  public boolean isScopeApi() {
 
-    return this.devon4jPackage.isLayerBatch();
+    return this.matcher.group(4).matches("api");
   }
 
-  public boolean isLayerClient() {
+  public boolean isScopeBase() {
 
-    return this.devon4jPackage.isLayerClient();
+    return this.matcher.group(4).matches("base");
   }
 
-  public String getScope() {
+  public boolean isScopeImpl() {
 
-    return this.devon4jPackage.getScope();
+    return this.matcher.group(4).matches("impl");
   }
 
-  public boolean isScopeApi() {
+  public String getApplication() {
 
-    return this.devon4jPackage.isScopeApi();
+    return this.matcher.group(5);
   }
 
-  public boolean isScopeBase() {
+  public String getComponent() {
 
-    return this.devon4jPackage.isScopeBase();
+    return this.matcher.group(3);
   }
 
-  public boolean isScopeImpl() {
+  public String getLayer() {
+
+    return this.matcher.group(2);
+  }
+
+  public String getRoot() {
 
-    return this.devon4jPackage.isScopeImpl();
+    return this.matcher.group(1);
   }
 
-  public String getDetail() {
+  public String getScope() {
 
-    return this.devon4jPackage.getDetail();
+    return this.matcher.group(4);
   }
 
-  public int hashCode() {
+  private List<String> getPatternGroups(String stringPattern) {
 
-    return this.devon4jPackage.hashCode();
+    List<String> patternGroups = new ArrayList<>();
+    Matcher m = Pattern.compile("\\((.*?)\\)").matcher(stringPattern);
+    while (m.find()) {
+      patternGroups.add(m.group(1));
+    }
+    return patternGroups;
   }
 
-  public boolean equals(Object obj) {
+  private void setPatternGroups() {
 
-    return this.devon4jPackage.equals(obj);
+    List<String> patternGroups = getPatternGroups(this.packages.getPattern());
+    this.layerPattern = Pattern.compile(patternGroups.get(1));
+    this.scopePattern = Pattern.compile(patternGroups.get(3));
   }
 
-  public String toString() {
+  private String replaceMappings(String packageName) {
 
-    return this.devon4jPackage.toString();
+    Map<String, String> mappings = this.packages.getMappings();
+    for (String key : mappings.keySet()) {
+      packageName = packageName.replace(key, mappings.get(key));
```

Comment:
we should not replace the mappings inside the regular expressions. Instead the idea is as following and I will stick to my example from #97:
```
"pattern": "([a-zA-Z0-9_]+\.)+(persistence|core|service|gui)\.([a-zA-Z0-9_]+)\.(api|base|impl)(\.[a-zA-Z0-9_]+)*",
        "groups": [ "root", "layer", "component", "scope", "detail" ],
        "mappings": {  "persistence": "dataaccess", "core": "logic", "gui": "client"}
```

Assuming we have `package` of `com.customer.myapp.persistence.mycomponent1.api.foo.bar` this will match the above `pattern` and result in the following groups:
`root`(1): `com.customer.myapp`
`layer`(2): `persistence`
`component`(3): `mycomponent1`
`scope`(4): `api`
`detail`(5): `foo.bar`

Now we like to assign this values but in order to be able to map this to `devon4j` standard architecture, we need to undertand what values such as `persistence` means for layer. That is where the `mappings` comes into play.
We use that `Map` to map each of these values. If there is no entry in that `Map`, we take the value as is. However, in our example the value `persistence` will now be mapped to `dataaccess` (what is a known devon4j layer so we can understand the architecture).

When you manipulate the regular expression by replacing it beforehand, it will not match the original pattern anymore. I hope the idea got clear now.

Reply:
I refactored the method so that the "old" value will still be inside the pattern and the package. E.g. `persistence` and `dataaccess` will now only be swapped for a comparison, e.g. in the method `isLayerDataAccess()`. 
Also nothing will be mapped if there is no entry in the Map for `persistence`.